src/pudl/etl.py

```diff
@@ -269,29 +300,46 @@ def etl_epacems(
             "Some timezones may be estimated based on plant state."
         )
 
-    # NOTE: This is a generator for raw dataframes
-    epacems_raw_dfs = pudl.extract.epacems.extract(
-        epacems_settings, Datastore(**ds_kwargs)
-    )
-
-    # NOTE: This is a generator for transformed dataframes
-    epacems_transformed_dfs = pudl.transform.epacems.transform(
-        epacems_raw_dfs=epacems_raw_dfs,
-        pudl_engine=pudl_engine,
-    )
-
     logger.info("Processing EPA CEMS data and writing it to Apache Parquet.")
     if logger.isEnabledFor(logging.INFO):
         start_time = time.monotonic()
 
-    # run the cems generator dfs through the load step
-    for df in epacems_transformed_dfs:
-        pudl.load.df_to_parquet(
-            df,
-            resource_id="hourly_emissions_epacems",
-            root_path=Path(pudl_settings["parquet_dir"]) / "epacems",
-            partition_cols=["year", "state"],
+    if epacems_settings.partition:
+        n_yrs = len(epacems_settings.years)
+        epacems_dir = Path(pudl_settings["parquet_dir"]) / "epacems"
+        with ProcessPoolExecutor() as executor:
+            _ = list(  # Convert results of map() to list to force execution
+                executor.map(
+                    _etl_one_year_epacems,
+                    epacems_settings.years,
+                    itertools.repeat(epacems_settings.states, n_yrs),
+                    itertools.repeat(pudl_settings["pudl_db"], n_yrs),
+                    itertools.repeat(epacems_dir, n_yrs),
+                    itertools.repeat(ds_kwargs, n_yrs),
+                )
+            )
+
+    else:
```

Comment:
I might be missing something but in this `else` part could you instead iterate through `epacems_settings.years` and call `_etl_one_year_epacems` instead of rewriting the logic?

Reply:
I wasn't thrilled about this duplication either, but as it's written right now the code is *slightly* different, because `_etl_one_year_epacems()` writes to a bunch of different files (one per year-state combo), creating a different `ParquetWriter` for each output, while the monolithic case uses the same `ParquetWriter` over and over again, writing each row-group into the same file. Am I missing an obvious way to refactor this and use the same code for both?

Reply:
Ah I forgot that in the single file case the same `ParquetWriter` is used repeatedly. 

The only way I really see using the same code for both is passing in the path for the `ParquetWriter` and having a different `ParquetWriter` open and write to a new state, year combo to that same file, which isn't great. Seems fine to leave the duplication then.

Reply:
I don't know how much additional overhead creating a new ParquetWriter every time would be, but yeah I guess that would work just fine!  I think this setup where we can do partitioned or monolithic is temporary -- we should settle on one or the other and just do it all the time... but I'm still trying to figure out which is preferable for the data catalog, So I think I'm inclined to leave it like this for now.